### PR TITLE
Update code and Dockerfiles for golang 1.17

### PR DESCRIPTION
**Pull request by yselkowitz**
- Fix gofmt for 1.17
- Update CI to golang-1.17

Closes #5364
Closes #5365
Closes #5368

### CODE REVIEW

File: data/assets.go [hunk 1]
```diff
@@ -1,4 +1,6 @@
+//go:build !release
 // +build !release
```
Comment (staebler) [inline data/assets.go:2]: Should we remove the `+build` directives, too, and force the move to go1.17 by changing https://github.com/openshift/installer/blob/cd170249fb5d7ce041ee35aae1486872cb076ef6/hack/build.sh#L8?

Comment (yselkowitz) [reply data/assets.go:2]: I think that's a bit premature.  For now I think it's sufficient to make sure installer *can* build with 1.17.

**Review by staebler — APPROVED** [review]
/lgtm

**Pull request merged**